src/Settings/Packages/Rts.hs

```diff
@@ -29,13 +38,67 @@ rtsLibffiLibraryName = do
         (False, False) -> "Cffi"
         (_    , True ) -> "Cffi-6"
 
+rtsLibffiIncludeArgs :: Args
+rtsLibffiIncludeArgs = package libffi ? builder (Ghc CompileCWithGhc) ? do
+  useSystemFfi <- expr $ flag UseSystemFfi
+  ffiIncludeDir <- getSetting FfiIncludeDir
+  mconcat [
+    useSystemFfi ? pure (map ("-I" ++) $ words ffiIncludeDir),
+    -- ffi.h triggers prototype warnings, so disable them here:
+    inputs [ "//Interpreter.c", "//Storage.c", "//Adjustor.c" ] ?
+    arg "-Wno-strict-prototypes" ]
+
 rtsLibffiLibrary :: Way -> Action FilePath
 rtsLibffiLibrary way = do
     name    <- rtsLibffiLibraryName
     suf     <- libsuf way
     rtsPath <- rtsBuildPath
     return $ rtsPath -/- "lib" ++ name ++ suf
 
+-- ref: mk/config.mk.in
+ghcRtsWithLibDw :: Action Bool
+ghcRtsWithLibDw = do
+    goodArch <- anyTargetArch ["i386", "x86_64"]
+    withLibDw <- setting WithLibDw
+    return $ goodArch && withLibDw == "YES"
```

Comment:
It looks like `WithLibDw` should be a `Flag`, not a `Setting`, so the comparison with `"YES"` is taken care of automatically.

Reply:
Oh I see, thanks!